lib/db/mysql.js

```diff
@@ -1040,6 +1102,36 @@ module.exports = function (log, error) {
       )
   }
 
+  MySql.prototype.writeMultiple = function (queries, resultHandler) {
+    return this.transaction(connection => {
+      return P.each(queries, (q, i) => {
+        const {sql, params} = q
+        return query(connection, sql, params)
+        .then(
+          function (result) {
+            log.trace('MySql.write', { sql, result })
+            if (resultHandler) {
+              return resultHandler(i, result)
```

Comment:
Just thinking out loud, ignore me if it's rubbish, but...

Instead of passing in one `resultHandler` and then giving it an index so it knows which query it's being invoked for, would it be a little bit cleaner if `resultHandler` was a property of the objects inside the `queries` array? So each query can have a dedicated result handler that doesn't need to check the value of `i`?